packages/apollo-server-core/src/requestPipeline.ts

```diff
@@ -364,11 +361,43 @@ export async function processGraphQLRequest<TContext>(
       requestContext as GraphQLRequestContextResponseForOperation<TContext>,
     );
     if (response == null) {
-      const executionDidEnd = await dispatcher.invokeDidStartHook(
+      // This execution dispatcher code is duplicated in `pluginTestHarness`
+      // right now.
+
+      const executionListeners: GraphQLRequestExecutionListener<TContext>[] = [];
+      dispatcher.invokeHookSync(
         'executionDidStart',
         requestContext as GraphQLRequestContextExecutionDidStart<TContext>,
+      ).forEach(executionListener => {
+        if (typeof executionListener === 'function') {
+          executionListeners.push({
+            executionDidEnd: executionListener,
+          });
+        } else if (typeof executionListener === 'object') {
+          executionListeners.push(executionListener);
+        }
+      });
+
+      const executionDispatcher = new Dispatcher(executionListeners);
+
+      // Create a callback that will trigger the execution dispatcher's
+      // `willResolveField` hook.  We will attach this to the context on a
+      // symbol so it can be invoked by our `wrapField` method during execution.
+      const invokeWillResolveField: GraphQLRequestExecutionListener<
+        TContext
+      >['willResolveField'] = (...args) =>
+          executionDispatcher.invokeDidStartHook('willResolveField', ...args);
+
+      Object.defineProperty(
```

Comment:
Do we want to skip defining the property and instrumenting the schema if there are no execution listeners? (Or specifically if there are no willResolveFields?) I suppose the old code always augments the schema with graphql-extensions even if there are no willResolveFields, so it's not a big deal either way.